app/src/main/java/com/github/se/polyfit/ui/components/button/BookmarkButton.kt

```diff
@@ -0,0 +1,45 @@
+package com.github.se.polyfit.ui.components.button
+
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.Done
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButtonDefaults
+import androidx.compose.material3.IconToggleButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import com.github.se.polyfit.model.recipe.Recipe
+
+@Composable
+fun BookmarkButton(
+    recipe: Recipe,
+    addRecipe: (Recipe) -> Unit = {},
+    removeRecipe: (Recipe) -> Unit = {},
+    modifier: Modifier = Modifier,
+) {
+  var isClicked by remember { mutableStateOf(false) }
```

Comment:
maybe for consistency with the `IconToggleButton`, a more suitable name would be `isChecked`

Reply:
Yep, way better naming. Thanks for the suggestion.